### PR TITLE
Docker registry

### PR DESCRIPTION
**Pull request by aviralsrivastava01**
Docker Registry

### CODE REVIEW

File: DockerUtil.go [hunk 1]
```diff
@@ -37,13 +33,56 @@ import (
...
-	log.Println(string(out))
-	waitForDockerDaemon(retryCount)
+func StartDockerDaemon(dockerConnection, dockerRegistryUrl, dockerCert string) {
+	connection := dockerConnection
+	fmt.Println(connection)
+	u, err := url.Parse(dockerRegistryUrl)
+	if err != nil {
+		log.Fatal(err)
+	}
+	if connection == "secure-with-cert" {
```
Comment (vikramdevtron) [inline DockerUtil.go:50]: use constants for static content "secure-with-cert"

File: DockerUtil.go [hunk 2]
```diff
@@ -37,13 +33,56 @@ import (
...
+
+		log.Println(string(out))
+		waitForDockerDaemon(retryCount)
+	} else if connection == "insecure" {
+		dockerdstart := "dockerd --insecure-registry " + dockerRegistryUrl + " --host=unix:///var/run/docker.sock --host=tcp://0.0.0.0:2375 > /usr/local/bin/nohup.out 2>&1 &"
+		out, _ := exec.Command("/bin/sh", "-c", dockerdstart).Output()
+		logStage("Insecure Registry")
+		log.Println(string(out))
+		waitForDockerDaemon(retryCount)
+	} else {
```
Comment (vikramdevtron) [inline DockerUtil.go:77]: can we remove this step and put this inside secure block

Comment (aviralsrivastava01) [reply DockerUtil.go:52]: Done Sir

File: DockerUtil.go [hunk 3]
```diff
@@ -37,13 +33,53 @@ import (
...
+
+	if connection == "insecure" {
+		dockerdstart := "dockerd --insecure-registry " + dockerRegistryUrl + " --host=unix:///var/run/docker.sock --host=tcp://0.0.0.0:2375 > /usr/local/bin/nohup.out 2>&1 &"
+		out, _ := exec.Command("/bin/sh", "-c", dockerdstart).Output()
+		logStage("Insecure Registry")
+		log.Println(string(out))
+		waitForDockerDaemon(retryCount)
+	} else {
+		if connection == "secure-with-cert" {
+			os.MkdirAll("/etc/docker/certs.d/"+dockerRegistryUrl, os.ModePerm)
```
Comment (vikramdevtron) [inline DockerUtil.go:54]: dockerRegistryUrl must be domain part only, without scheme.

File: DockerUtil.go [hunk 4]
```diff
@@ -37,13 +33,53 @@ import (
...
+			if err2 != nil {
+				log.Fatal(err2)
+			}
+			dockerdStart := "dockerd --host=unix:///var/run/docker.sock --host=tcp://0.0.0.0:2375 > /usr/local/bin/nohup.out 2>&1 &"
+			out, _ := exec.Command("/bin/sh", "-c", dockerdStart).Output()
+			logStage("Secure with Cert")
+
+			log.Println(string(out))
+			waitForDockerDaemon(retryCount)
+		} else {
```
Comment (vikramdevtron) [inline DockerUtil.go:74]: duplicate code

**Comment by vikramdevtron** [comment]
not in use deprecated

**Pull request closed without merging**